strax/processor.py

```diff
@@ -156,12 +157,21 @@ def __init__(self,
                         executor=executor),
                     name=f'divide_outputs:{d}')
 
+                # If we have a plugin with double dependency both outputs
+                # of a multioutput-plugin are required. Hence flow-freely
+                # is empty an needs to be updated here:
+                provided_data_types = set(p.provides)
+                reader_data_types = set(strax.to_str_tuple(d))
+                _flow_freely = (to_flow_freely | ((provided_data_types - reader_data_types) -
+                                                  to_flow_freely))
```

Comment:
Please make sure if this is correct, but it looks like this could be a bit simpler, right?
```suggestion
                double_dependency = (provided_data_types - reader_data_types)
                to_flow_freely |= double_dependency 
```

Reply:
you'd also have to change the _flow_freely a few lines down